packages/vc-export/src/KiltCredentialV1.ts

```diff
@@ -0,0 +1,365 @@
+    type: DEFAULT_CREDENTIAL_TYPES,
+    ...(id && { id }),
+    nonTransferable: true,
+    credentialSubject,
+    credentialSchema,
+    issuer,
+    issuanceDate,
+    ...(claimHash && {
+      credentialStatus: fromGenesisAndRootHash(chainGenesisHash, claimHash),
+    }),
+    ...(federatedTrustModel.length > 0 && { federatedTrustModel }),
+  }
+}
+
+export const credentialSchema: JsonSchema.Schema = {
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  type: 'object',
+  properties: {
+    '@context': {
+      type: 'array',
+      const: DEFAULT_CREDENTIAL_CONTEXTS,
+    },
+    type: {
+      type: 'array',
+      uniqueItems: true,
+      minItems: 2,
+      maxItems: 2,
+      oneOf: [
+        { items: { enum: DEFAULT_CREDENTIAL_TYPES } },
+        {
+          items: {
+            enum: [
+              `${W3C_CREDENTIAL_CONTEXT_URL}#${W3C_CREDENTIAL_TYPE}`,
+              `${KILT_CREDENTIAL_CONTEXT_URL}#${KILT_CREDENTIAL_TYPE}`,
+            ],
+          },
+        },
+      ],
+    },
+    id: {
+      type: 'string',
+      format: 'uri',
+    },
+    nonTransferable: {
+      type: 'boolean',
+    },
+    credentialSubject: {
+      type: 'object',
+      properties: {
+        id: {
+          type: 'string',
+          format: 'uri',
+        },
+      },
+      required: ['id'],
+    },
+    issuer: {
+      type: 'string',
+      format: 'uri',
+    },
+    issuanceDate: {
+      type: 'string',
+      format: 'date-time',
+    },
+    credentialStatus: {
+      type: 'object',
+      properties: {
+        id: {
+          type: 'string',
+          format: 'uri',
+        },
+        type: {
+          // public credentials may have a different revocation check, so we don't force the type here
+          type: 'string',
+        },
+      },
+      required: ['id', 'type'],
+    },
+    federatedTrustModel: {
+      type: 'array',
+      minLength: 1,
+      items: {
+        type: 'object',
+        properties: {
+          id: {
+            type: 'string',
+            format: 'uri',
+          },
+          type: {
+            type: 'string',
+          },
+        },
+      },
+    },
+    credentialSchema: {
+      type: 'object',
+      properties: {
+        id: {
+          type: 'string',
+          format: 'uri',
+        },
+        type: {
+          type: 'string',
+          const: JSON_SCHEMA_TYPE,
+        },
+        schema: { $ref: CType.Schemas.CTypeModel.$id },
+      },
+      required: ['id', 'type'],
+    },
+    proof: {
+      type: 'object',
+      properties: {
+        type: {
+          type: 'string',
+        },
+      },
+      required: ['type'],
+    },
+  },
+  additionalProperties: false,
+  required: [
+    '@context',
+    'type',
+    'id',
+    'credentialSubject',
+    'issuer',
+    'issuanceDate',
+    'credentialStatus',
+    'credentialSchema',
+  ],
+}
+
+const schemaValidator = new JsonSchema.Validator(credentialSchema, '7')
```

Comment:
> Maybe pull `'7'` out to a const somewhere as well?

 Actually I think we can drop the 7. The JSON schema version should be inferred from the $schema property of the schema.

> Plus, do we instantiate it outside of the function for caching? I'm not super familiar with how and when this would be called. Once the module is imported? Every time it is imported?

Yes, this is run on import, once, instead of being run every time the function is called.

Reply:
I see it used in a few places. Can we pull it out in a const then?

Reply:
Or perhaps some sort of builder that would return the right object given the schema definition.

Reply:
> This value always depends on the schema version used by the schema defined in the same file; a global const seems misplaced here.

Is there any occurrence in which, within the same file, you might want to create definitions with different versions? I did not mean a global constant, but just a way to set it once, and simply use it, even within the same file.